### PR TITLE
Use `id` as key instead of `elector`.

### PR DESCRIPTION
**Pull request by mattcollier**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ api.dereferenceElectorPool = async ({electorPool, ledgerNode}) => {
   for(const e of electorPool) {
     // service may be a string referencing a serviceId contained in a DID
     // Document referenced in `e.elector`
-    if(typeof e.service === 'string') {
+    if(typeof e.service === 'string' && typeof e.elector === 'string') {
```
Comment (dlongley) [inline lib/index.js]: Why this change? Why not just use `e.id` and use that for the record lookup as well? I don't really understand the purpose of `elector` at all at this point.

**Review by dlongley — APPROVED** [review]
Approving whether not anything is done with my comment.

File: lib/index.js (hunk 1 above)
Comment (mattcollier) [reply lib/index.js]: The idea *was* that electors could have DID documents that included service descriptors.  So `elector` would contain a DID.

Comment (dlongley) [reply lib/index.js]: But the `service` property is on the object with `id` and `elector` not *in* `elector` as an object, right? That means `id` should be a DID if we want to support that.

We should have either:

```js
{
  id: "<whatever>",
  // needs dereferencing
  service: "<service ID>"
}
```

OR

```js
{
  id: "<whatever>",
  service: {
    /* serviceEndpoint etc */
  }
}
```

OR

```js
{
  id: "<whatever>",
  elector: {
    id: "<did/whatever>",
    // needs dereferencing
    service: "<service ID>"
  }
}
```

OR

```js
{
  id: "<whatever>",
  elector: {
    id: "<did/whatever>",
    service: {
      /* serviceEndpoint, etc. */
    }
  }
}
```

But right now we have some weird mishmash.

Comment (dlongley) [reply lib/index.js]: In short, we should just do the simplest two options:

```js
{
  id: "<whatever>",
  // needs dereferencing (first part of service ID is a DID)
  service: "<service ID>"
}

OR

{
  id: "<whatever>",
  // embedded case
  service: {
    /* id, serviceEndpoint etc */
  }
}
```

File: lib/index.js [hunk 2]
```diff
@@ -27,11 +27,14 @@ api.dereferenceElectorPool = async ({electorPool, ledgerNode}) => {
 
   for(const e of electorPool) {
     // service may be a string referencing a serviceId contained in a DID
-    // Document referenced in `e.elector`
     if(typeof e.service === 'string') {
+      // extract the DID from `service` which is a DID followed by a fragment
+      const recordId = e.service.substr(0, e.service.indexOf('#'));
```
Comment (dlongley) [inline lib/index.js:32]: As long as a validator ensures the `#` is there, we're good here.

**Review by dlongley — APPROVED** [review]
LGTM

File: lib/index.js (hunk 2 above)
Comment (mattcollier) [reply lib/index.js:32]: right, currently being validated as a string, I'll add an issue in validator.
